pydantic/dataclasses.py

```diff
@@ -143,9 +143,29 @@ def dataclass(
 
     if sys.version_info >= (3, 10):
         kwargs = dict(kw_only=kw_only, slots=slots)
+
+        def make_pydantic_fields_compatible(cls: type[Any]) -> None:
+            """Make sure that stdlib `dataclasses` understands `Field` kwargs like `kw_only`
+            To do that, we simply change
+              `x: int = pydantic.Field(..., kw_only=True)`
+            into
+              `x: int = dataclasses.field(default=pydantic.Field(..., kw_only=True), kw_only=True)`
+            """
+            for field_name in cls.__annotations__:
+                try:
+                    field_value = getattr(cls, field_name)
+                except AttributeError:
+                    # no default value has been set for this field
+                    continue
+                if isinstance(field_value, FieldInfo) and field_value.kw_only:
```

Comment:
For now I went with the smallest change possible instead of wrapping them all the time and forward other stuff like `init`. I suggest we wait to see if this method needs to more than that